### PR TITLE
Ops average

### PR DESCRIPTION
**Pull request by ellesmith88**
### Pull Request Checklist:
- This PR adds `average_over_dims`
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] I have added my relevant user information to `AUTHORS.md`

* **What kind of change does this PR introduce?:** 
Adds `average_over_dims` in `clisops.core` and `clisops.ops` and restructures the operations within `clisops.ops` to use a base operation.

I have removed the checkbox in the PR template to call bumpversion minor as we haven't really been using this as [art of our versioning/release procedure

Requires https://github.com/roocs/roocs-utils/pull/55

* **Does this PR introduce a breaking change?:** 
No just adds `average_over_dims`

* **Other information:** 
See discussion in PR #109 for resolving conflicts with `average_shape` @aulemahal please review this PR as well

### CODE REVIEW

File: clisops/ops/subset.py
Comment (Zeitsperre) [inline clisops/ops/subset.py:151]:
```suggestion
        A list of the subsetted outputs in the format selected; str corresponds to file paths if the
        output format selected is a file.
```

File: clisops/ops/subset.py
Comment (Zeitsperre) [inline clisops/ops/subset.py:141]:
```suggestion
    level: Optional[Union[str, Tuple[Union[int, float, str], Union[int, float, str]], LevelParameter]
```

File: clisops/ops/average.py
Comment (Zeitsperre) [inline clisops/ops/average.py:5]: I've been trying to run the tests but the `roocs-utils` I have (even with master) doesn't seem to have this module, so tests are failing at the collection step.

File: clisops/ops/subset.py
Comment (Zeitsperre) [reply clisops/ops/subset.py:141]: This needs to stay on one line, otherwise it will be treated as flavour text for the call signature.

**Review by Zeitsperre — COMMENTED** [review]
I'm not sure if I can approve as I haven't worked much on the averaging code. The new design patterns (classes for parsing ops requests) looks good. 

I'm not able to run the tests due to a missing module in `roocs-utils`. Is it possible that the version of `roocs-utils` being used here is not the same as the one on `master`?

I would invite @aulemahal to weigh in on this if he's available.

**Comment by ellesmith88** [comment]
@Zeitsperre Yes sorry - I mentioned that it requires https://github.com/roocs/roocs-utils/pull/55. This hasn't been merged yet as it will break the file naming in the current `clisops` master.

Once the averaging is in I'll release `roocs-utils` then `clisops` so that `clisops` isn't dependent on the master branch of `roocs-utils` any more.

Thanks for looking through - It would be great to have a review from @aulemahal (Couldn't request it through github!)

**Review by agstephens — APPROVED** [review]
Hi @Zeitsperre and @aulemahal, I am happy with these proposed changes. We will need to do some integration with @aulemahal's branch soon - but I suggest we merge this one first.

**Comment by Zeitsperre** [comment]
@agstephens @ellesmith88 Once this is merged, @aulemahal tells me will be able to integrate it into his branch to finalize the work there. So I guess the priority is finish #125 → finish #109 → release?

**Comment by ellesmith88** [comment]
@agstephens I can go ahead and merge the `roocs-utils` branch needed for this then this PR, I think we're happy that this is finished?

**Pull request merged**
